pkg/storage/unified/search/bleve.go

```diff
@@ -554,6 +551,42 @@ func toBleveSearchRequest(req *resource.ResourceSearchRequest, access authz.Acce
 		searchrequest.Query = bleve.NewConjunctionQuery(queries...) // AND
 	}
 
+	// Can we remove this? Is access ever nil?
+	if access != nil {
+		auth, ok := claims.From(ctx)
+		if !ok {
+			return nil, resource.AsErrorResult(fmt.Errorf("missing claims"))
+		}
+		checker, err := access.Compile(ctx, auth, authz.ListRequest{
+			Namespace: b.key.Namespace,
+			Group:     b.key.Group,
+			Resource:  b.key.Resource,
+			Verb:      utils.VerbList,
+		})
+		if err != nil {
+			return nil, resource.AsErrorResult(err)
+		}
+		checkers := map[string]authz.ItemChecker{
+			b.key.Resource: checker,
+		}
+
+		// handle federation
+		for _, federated := range req.Federated {
+			checker, err := access.Compile(ctx, auth, authz.ListRequest{
+				Namespace: federated.Namespace,
+				Group:     federated.Group,
+				Resource:  federated.Resource,
+				Verb:      utils.VerbList,
+			})
+			if err != nil {
+				return nil, resource.AsErrorResult(err)
+			}
+			checkers[federated.Resource] = checker
```

Comment:
*technically* not guaranteed unique -- but I think totally fine for now.  (unique would be group+resource)